app/javascript/onboarding/__tests__/Onboarding.test.jsx

```diff
@@ -311,18 +312,32 @@ describe('<Onboarding />', () => {
       );
     });
 
+    afterEach(() => {
+      window.location = location;
+    });
+
     test('renders properly', () => {
       expect(onboardingSlides).toMatchSnapshot();
     });
 
-    test('should allow user to advance', async () => {
+    test('should redirect user when finished', async () => {
       fetch.once({});
 
+      // Setup: Enable window.location to be writable.
+      const url = 'https://dummy.com/onboarding';
+      Object.defineProperty(window, 'location', {
+        value: { href: url },
+        writable: true,
+      });
+
+      expect(window.location.href).toBe(url);
       onboardingSlides.find('.next-button').simulate('click');
       await flushPromises();
+      // No longer advance slide.
       expect(onboardingSlides.state().currentSlide).toBe(
-        emailPreferencesFormIndex + 1,
+        emailPreferencesFormIndex,
       );
+      expect(window.location.href).toBe('/');
```

Comment:
This seems odd; why does the `onboardingSlides.state().currentSlide` still point to the `emailPreferencesFormIndex`, while the `window.location.href` is also `/`? Seems like both of them cannot be true at the same time?

Reply:
I'm not too sure what context the Jest test environment has because technically there shouldn't be any more onboardingSlides due to the redirect. To check if `window.location.href` change is essential but making sure the slide number of the state didn't change is just to account for unintended behavior.